### DISCUSSION TITLE
get_images always get all images in whole PDF doucument

### DISCUSSION THREAD

**Discussion by yonglee7015** [pymupdf/PyMuPDF#2126] [Q&A] [answered]
get_images  ---- I already set page = first page  such as pdf =pdf_file[0]. But it still get all images in whole pdf document.

**Comment by JorjMcKie** [comment, marked as answer]
A PDF creator has the liberty to define **_one_** `/Resources` object in the file, which refers to **_all_** the images and fonts and other stuff, which furtheron may be used by pages. Of course, a single page will only use a subset.
To build a page-specific subset, execute `page.clean_contents()` before doing `page.get_images()` etc.

**Reply by yonglee7015** [reply]
page.clean_contents(), it works. Thanks for your kindly help.
